### PR TITLE
Implement try-parse like pattern

### PR DESCRIPTION
**Pull request by kkjasoncheung**
- Refactor code to use a `try-parse` like pattern to remove redundant code.
- Renames some variables for consistency

### CODE REVIEW

File: stronk/schemas/program/type.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ class Meta:
     program_reviews = graphene.List(ProtectedProgramReviews)
 
     def resolve_author(root, info):
-        return UserModel.find_by_id(root.author)
+        return UserModel.try_find_by_id(root.author)
```
Comment (boiledbuns) [inline stronk/schemas/program/type.py]: I noticed we're `try_xxx` here. Is there a guideline when we should use `try_xxx` vs just `xxx` here? I think there's a constraint on author field st the user will always exist

**Review by boiledbuns — APPROVED** [review]
LGTM: Just two things I noticed:
1. We seem to be using the `try_parse` pattern a bit differently from how it's described [here](https://stackoverflow.com/questions/182440/tryparse-parse-like-pattern-what-is-the-best-way-to-implement-it) which specifically prevents errors from being thrown in the `try_xxx` methods but lets them through for the normal `xxx` methods. Here however, we we bubble up our own error for the `xxx` methods.

For example: 
```python
class User(db.Model):
    ...

    @staticmethod
    def try_find_by_id(id: str):
        return User.query.filter_by(id=id).first()

    @staticmethod
    def find_by_id(id: str):
        """Returns the user with id. Raises a NotFound exception
        if the user does not exist.
        """
        user = User.try_find_by_id(id)
        if not user:
            raise NotFound(USER_NOT_FOUND_MSG)
        return user

    ...
}
```
I've only learned about this pattern recently so lmk if there's something I'm missing, but it feels like we're creating two ways to access the data where one of them is possibly empty (try_find_by_id) and the other is not (find_by_id). This is just something I noticed and wanted to make sure I was understanding correctly so no need to change anything. 

2. Another thing I noticed is that not all non try methods do the same thing. For example,  the `get_by_reviewer_and_program_id` which is being used in `DeleteProgramReview` doesn't raise the not found error. I hope I'm not missing anything here, but should we be keeping the naming scheme consistent?

**Comment by kkjasoncheung** [comment]
@boiledbuns 
To address your points:
1. It is true that we are creating two ways to access the data where one of them is possibly empty (like `try_find_by_id`) and the other one can also be empty, but will raise an exception is this case (`find_by_id`). The reason for this is because our application doesn't always need to throw an exception when we don't get data back.

For instance in `stronk/schemas/user/type.py` if the user doesn't have a program, `resolve_currentProgram` should just return empty rather than raise an exception.
```
 def resolve_currentProgram(root, info):
        # raise an exception
        return ProgramModel.find_by_id(root.current_program)
```
vs
```
 def resolve_currentProgram(root, info):
        # does not raise exception and graphene will convert this to `null` in the response body
        return ProgramModel.try_find_by_id(root.current_program)
```

2. It's expected that not all non try methods do the same thing, but all `try` methods should do the same thing. The convention is to only include a `try` variant for methods where the non-try variant is prone to raising an exception.

File: stronk/schemas/program/type.py (hunk 1 above)
Comment (kkjasoncheung) [reply stronk/schemas/program/type.py]: @boiledbuns oo true good catch. A good guideline would be ensuring that we only use `try_xxx` when there isn't a constraint saying the field must be non-nullable

**Review by MrRDotS — APPROVED** [review]

**Pull request merged**